wolfcrypt/src/rsa.c

```diff
@@ -4441,12 +4455,35 @@ int wc_MakeRsaKey(RsaKey* key, int size, long e, WC_RNG* rng)
 #ifdef WOLF_CRYPTO_CB
     if (key->devId != INVALID_DEVID) {
         err = wc_CryptoCb_MakeRsaKey(key, size, e, rng);
+    #ifndef WOLF_CRYPTO_CB_ONLY_RSA
         if (err != CRYPTOCB_UNAVAILABLE)
             goto out;
         /* fall-through when unavailable */
+    #else
+        #ifdef WOLFSSL_SMALL_STACK
+        if (p)
```

Comment:
A `goto out` will free these. I recommend just doing a goto out instead. You'll need to include the goto out section below. Then we aren't duplicating this code.

Reply:
Thanks. Included the goto out section for free.